src/NUnitFramework/framework/Internal/Builders/DefaultTestCaseBuilder.cs

```diff
@@ -132,11 +132,30 @@ public Test BuildFrom(FixtureMethod method, Test suite)
                     tests.Add(test); 
             }
 
-            return hasBuildersSpecified && method.Method.GetParameters().Length > 0 || tests.Count > 0
+            return hasBuildersSpecified && method.Method.GetParameters().Length > 0 || 
+                   tests.Count > 0 ||
+                   MethodHasDataAttributeParameters(method)
                 ? BuildParameterizedMethodSuite(method, tests)
                 : BuildSingleTestMethod(method, suite);
         }
 
+        /// <summary>
+        /// Checks to see if the method has arguments with DataAttribute, 
+        /// if the argument has no data, the test should not run rather than fail.
+        /// </summary>
+        /// <param name="fixtureMethod">The method for which a test is to be built.</param>
+        /// <returns>True if the method signature contains DataAttribute arguments</returns>
+        private bool MethodHasDataAttributeParameters(FixtureMethod fixtureMethod)
+        {
+            foreach (var parameter in fixtureMethod.Method.GetParameters())
+            {
+                var dataAttributes = parameter.GetAttributes<DataAttribute>(true);
+                if (dataAttributes.Length > 0)
+                    return true;
```

Comment:
You are returning true if __any__ parameters have a parameter data source specified. Is that what we want in the case where the user has mistakenly added a source to one parameter and not to another? I haven't worked out the logic of this, but we want to be sure that an error message is still given in that case.

Reply:
@DaiMichael Also, these lines could be refactored to use `HasAttribute` rather than `GetAttributes….Length != 0`.